flake.nix

```
@@ -12,25 +12,37 @@
     nixpkgs,
     ...
   } @ inputs: let
+    # We should only specify the modules Hjem explicitly supports, or we risk
+    # allowing not-so-defined behaviour. For example, adding nix-systems should
+    # be avoided, because it allows specifying Hjem is not tested on.
     forAllSystems = nixpkgs.lib.genAttrs ["x86_64-linux" "aarch64-linux"];
   in {
     nixosModules = {
       hjem = ./modules/nixos;
       default = self.nixosModules.hjem;
     };
 
+    packages = forAllSystems (system: {
+      # Expose the 'smfh' instance used by Hjem as a package in the Hjem flake
+      # outputs. This allows consuming the exact copy of smfh used by Hjem.
+      smfh = inputs.smfh.packages.${system}.smfh;
```

```suggestion
      linker = inputs.smfh.packages.${system}.smfh;
```

Shouldn't this be linker instead? So that it makes it more clear what `smfh` is to someone who is not familiar with our project. This would also make sense if we do decide to go with linker as the input name too (which would potentially prove extremely useful if one were to want to check their in-development linker against our flake checks)

---

Since the flake provides an input (and therefore a link), I don't think there is an ambiguity. Package meta is also available, so description and homepage are still visible. A user hoping to work with our linker will need to visit smfh's page *anyway*, so there is no need imo.